src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -132,9 +139,19 @@ protected override Task OnClientConnectedAsync(OpenConnectionMessage message)
             {
                 isDiagnosticClient = Convert.ToBoolean(isDiagnosticClientValue.FirstOrDefault());
             }
+            await Task.Yield();
+
+            var diagnosticLogsContext = new DiagnosticLogsContext();
 
-            using (new ClientConnectionScope(outboundConnection: this, isDiagnosticClient: isDiagnosticClient))
+            using (new ClientConnectionScope(outboundConnection: this, isDiagnosticClient: isDiagnosticClient, isServiceEnableMessageLog: diagnosticLogsContext))
             {
+                if (HubEndpoint != null)
+                {
+                    HubEndpoint.EnableMessageLogEventHandler += enableMessageLog =>
```

Comment:
>EnableMessageLogEventHandler [](start = 32, length = 28)

any other way instead of adding event handler to `HubEndpoint`, we tend to make the class hosting endpoint properties simple and clean

Reply:
I think I can try create a new scope in service connection container level, then all the pub sub events work will be done in the same container, then no need to pass the info to service connection. Then move the event handler from `ServiceHubEndpoint` to `ServiceConnectionContainerBase` as a private field.

Btw, the current implementation is based on that we need to update `ClientConnectionScope` (inside `ServiceConnection`) and `HubServiceEndpoint` is the only suitable structure passes from container to connection. If we still use `ClientConnectionScope` we need to modify `IServiceConnectionFactory` to pass the new structure.